src/projections/polyfill-colspan.js

```diff
@@ -0,0 +1,87 @@
+import { last } from '../utils';
+
+function processRow({ tds = [] }, rowSpans) {
+  let start = 0;
+  const retRowSpans = rowSpans;
+  tds.forEach((td) => {
+    while (retRowSpans[start] > 0) {
+      retRowSpans[start] -= 1;
+      start += 1;
+    }
+
+    const newStart = start + (td.props.colspan || 1);
+    while (start < newStart) {
+      if (retRowSpans[start] > 0) {
+        retRowSpans[start] -= 1;
+      }
+
+      if (td.props.rowspan > 0) {
+        retRowSpans[start] = Math.max(td.props.rowspan - 1, retRowSpans[start] || 0);
+      }
+      start += 1;
+    }
+  });
+
+  return {
+    colNum: start,
+    rowSpans: retRowSpans,
+  };
+}
+
+function processSection({ trs = [] }) {
```

Comment:
You may pass in a `context` which contains the `maxWidth` and the `lastTds`. So that you can
1. Calculate the `maxWidth` on the fly
2. Collect only the `colspan == 0` cells